plugin/storage/integration/integration_test.go

```diff
@@ -0,0 +1,276 @@
+	reader spanstore.Reader
+
+	// cleanUp() should ensure that the storage backend is clean before another test.
+	// called either before or after each test, and should be idempotent
+	cleanUp func() error
+
+	// refresh() should ensure that the storage backend is up to date before being queried.
+	// called between set-up and queries in each test
+	refresh func() error
+}
+
+// QueryFixtures and TraceFixtures are under ./fixtures/queries.json and ./fixtures/traces/*.json respectively.
+// Each query fixture includes:
+// 	Caption: describes the query we are testing
+// 	Query: the query we are testing
+//	ExpectedFixture: the trace fixture that we want back from these queries.
+// Queries are not necessarily numbered, but since each query requires a service name,
+// the service name is formatted "query##-service".
+type QueryFixtures struct {
+	Caption         string
+	Query           *spanstore.TraceQueryParameters
+	ExpectedFixture string
+}
+
+func (s *StorageIntegration) IntegrationTestGetServices(t *testing.T) {
+	t.Log("Testing GetServices ...")
+
+	expected := []string{"example-service-1", "example-service-2", "example-service-3"}
+	s.getBasicFixture(t)
+
+	s.refresh()
+
+	var found bool
+
+	var actual []string
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err := s.reader.GetServices()
+		require.NoError(t, err)
+		if found = assert.ObjectsAreEqualValues(expected, actual); found {
+			break
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetOperations(t *testing.T) {
+	t.Log("Testing GetOperations ...")
+
+	expected := []string{"example-operation-1", "example-operation-3", "example-operation-4"}
+	s.getBasicFixture(t)
+
+	s.refresh()
+
+	var found bool
+	var actual []string
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err := s.reader.GetOperations("example-service-1")
+		require.NoError(t, err)
+		if found = assert.ObjectsAreEqualValues(expected, actual); found {
+			break
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetTrace(t *testing.T) {
+	t.Log("Testing GetTrace ...")
+
+	expected := s.getBasicFixture(t)
+	expectedTraceID := model.TraceID{Low: 17}
+
+	s.refresh()
+
+	var found bool
+	var actual *model.Trace
+	for i := 0; i < iterations; i++ {
+		s.logger.Info(fmt.Sprintf(waitForBackendComment, i+1, iterations))
+		actual, err := s.reader.GetTrace(expectedTraceID)
+		if err == nil {
+			if found = len(actual.Spans) == len(expected.Spans); found {
```

Comment:
you don't need the found variable, you can just do the length check here and L148